digiroad2-geo/src/test/scala/fi/liikennevirasto/digiroad2/lane/LaneFillerSpec.scala

```diff
@@ -0,0 +1,313 @@
+    filledTopology.map(_.sideCode) should be (Seq(SideCode.TowardsDigitizing.value,  SideCode.TowardsDigitizing.value, SideCode.AgainstDigitizing.value))
+    filledTopology.map(_.id) should be (Seq(20L, 21L, 0L))
+    filledTopology.map(_.linkId) should be (Seq( 2L, 2L, 2L))
+    filledTopology.map(_.geometry) should be (Seq(Seq(Point(0.0, 0.0), Point(10.0, 0.0)), Seq(Point(4.0, 0.0), Point(10.0, 0.0)), Seq(Point(0.0, 0.0), Point(15.0, 0.0)) ))
+
+    changeSet.expiredLaneIds should be (Set(22L))
+  }
+
+  test("Drop lanes with less than 2 meters") {
+    val topology = Seq( roadLinkTowards1 )
+
+    val lane = PersistedLane(20L, roadLinkTowards1.linkId, SideCode.TowardsDigitizing.value,
+      11, 745L, 1.0, 2.0, None, None, None, None, expired = false, 0L, None,
+      LanePropertiesValues(Seq(LaneProperty("lane_code", Seq(LanePropertyValue(11)))))
+    )
+
+    val lane12 = PersistedLane(21L, roadLinkTowards1.linkId, SideCode.TowardsDigitizing.value,
+      12, 745L, 8.0, 10.0, None, None, None, None, expired = false, 0L, None,
+      LanePropertiesValues(Seq(LaneProperty("lane_code", Seq(LanePropertyValue(12)))))
+    )
+
+
+    val linearAssets = Map(
+      roadLinkTowards1.linkId -> Seq( lane, lane12 )
+    )
+
+    val (filledTopology, changeSet) = laneFiller.fillTopology(topology, linearAssets)
+
+    filledTopology should have size 2
+    filledTopology.map(_.sideCode) should be (Seq(SideCode.TowardsDigitizing.value, SideCode.TowardsDigitizing.value ))
+    filledTopology.map(_.id) should be (Seq(21L, 0L))
+    filledTopology.map(_.linkId) should be (Seq( 1L, 1L))
+    filledTopology.map(_.geometry) should be (Seq(Seq(Point(8.0, 0.0), Point(10.0, 0.0)), Seq(Point(0.0, 0.0), Point(10.0, 0.0)) ))
+
+    changeSet.expiredLaneIds should be (Set(20L))
+  }
+
+  test("Don't drop lanes with less than 2 meters on a roadLink with length less than 2 meters") {
+    val topology = Seq( roadLinkTowards4 )
+
+    val lane = PersistedLane(20L, roadLinkTowards4.linkId, SideCode.AgainstDigitizing.value,
+      21, 745L, 0.0, 1.9, None, None, None, None, expired = false, 0L, None,
+      LanePropertiesValues(Seq(LaneProperty("lane_code", Seq(LanePropertyValue(21)))))
+    )
+
+    val lane12 = PersistedLane(21L, roadLinkTowards4.linkId, SideCode.AgainstDigitizing.value,
+      22, 745L, 0.0, 1.0, None, None, None, None, expired = false, 0L, None,
+      LanePropertiesValues(Seq(LaneProperty("lane_code", Seq(LanePropertyValue(22)))))
+    )
+
+
+    val linearAssets = Map(
+      roadLinkTowards4.linkId -> Seq( lane, lane12 )
+    )
+
+    val (filledTopology, changeSet) = laneFiller.fillTopology(topology, linearAssets)
+
+    filledTopology should have size 2
+    filledTopology.map(_.sideCode) should be (Seq(SideCode.AgainstDigitizing.value, SideCode.AgainstDigitizing.value ))
+    filledTopology.map(_.id) should be (Seq(20L, 21L))
+    filledTopology.map(_.linkId) should be (Seq( 4L, 4L))
+
+    changeSet.expiredLaneIds should be (Set())
+  }
+
```

Comment:
I think it's missing a Unit test for the case when road link modifies to TowardsDigitizing/AgainstDigitizing and have both Lanes (BothDirection). The missed lane should be expired.

Reply:
Done